pkg/spoke/submarineragent/config_controller.go

```diff
@@ -86,6 +88,7 @@ func NewSubmarinerConfigController(input *SubmarinerConfigControllerInput) facto
 		clusterName:          input.ClusterName,
 		cloudProviderFactory: input.CloudProviderFactory,
 		onSyncDefer:          input.OnSyncDefer,
+		configName:           input.configName,
```

Comment:
How is the caller to determine the `SubmarinerConfig` name on startup? Even if you query for it, it may not exist yet.